apps/barrel_httpc/src/barrel_httpc.erl

```diff
@@ -242,6 +271,27 @@ put(Conn, #{ <<"id">> := DocId } = Doc, Options0) ->
   post_put(Conn, <<"PUT">>, Doc, Url, Headers);
 put(_, _, _) -> erlang:error({bad_doc, invalid_docid}).
 
+%% @doc create or update a document with attachments
+-spec put(Conn, Doc, Attachments, Options) -> Res when
+    Conn::conn(),
+    Doc :: doc(),
+    Attachments :: [attachment()],
+    Options :: write_options(),
+    Res :: {ok, docid(), rev()} | {error, conflict} | {error, any()}.
+put(Conn, #{ <<"id">> := DocId } = Doc, Attachments, Options0) ->
+  {Headers, Options1} = headers(Options0),
+  Url = barrel_httpc_lib:make_url(Conn, [<<"docs">>, DocId], Options1),
+  case encode_attachments(Attachments) of
```

Comment:
i would rather pass this option to the `post_put` function to not duplicate the code. To make the code simpler this function could have this signature:

```erlang
post_put(Connection, Method, Doc, Url, Headers, Attachments).
```

If `Attachments` is an empty list (for `put/3` and `post/3`) the processing an attachment is ignored, either way it would be encoded.